### PR TITLE
pythonPackages.pysimplegui: init at 4.55.1

### PR DESCRIPTION
**Pull request by lucasew**
Signed-off-by: lucasew <lucas59356@gmail.com>

###### Motivation for this change
- This lib provides a way to make GUIs easily in Python, thing that will be useful for NixOS mass adoption

###### Things done
- Package PySimpleGUI

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by thiagokokada** [comment]
Please fix the commit message:

```
pythonPackages.pysimplegui: init at 4.55.1
```

File: pkgs/development/python-modules/pysimplegui/default.nix [hunk 1]
```diff
@@ -0,0 +1,32 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+, python3Packages
+, pkgs
+, fetchPypi
+, tkinter
+, xlib
+, xvfb-run
```
Comment (thiagokokada) [inline pkgs/development/python-modules/pysimplegui/default.nix:9]: Not used.

Comment (lucasew) [reply pkgs/development/python-modules/pysimplegui/default.nix]: removed

File: pkgs/development/python-modules/pysimplegui/default.nix [hunk 2]
```diff
@@ -0,0 +1,30 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+, python3Packages
+, pkgs
```
Comment (thiagokokada) [inline pkgs/development/python-modules/pysimplegui/default.nix:5]: Not used too.
```suggestion
```

File: pkgs/development/python-modules/pysimplegui/default.nix [hunk 3]
```diff
@@ -0,0 +1,30 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
```
Comment (thiagokokada) [inline pkgs/development/python-modules/pysimplegui/default.nix:3]: Not used too.
```suggestion
```

File: pkgs/development/python-modules/pysimplegui/default.nix [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+, python3Packages
```
Comment (thiagokokada) [inline pkgs/development/python-modules/pysimplegui/default.nix:4]: Not used too.
```suggestion
```

File: pkgs/development/python-modules/pysimplegui/default.nix (hunk 2 above)
Comment (lucasew) [reply pkgs/development/python-modules/pysimplegui/default.nix]: solved

**Review by thiagokokada — COMMENTED** [review]
Is there some simple way to test it?

CC @lucasew.

**Comment by lucasew** [comment]
> Is there some simple way to test it?
> 
> CC @lucasew.

- cd to the nixpkgs root
- `NIX_PATH=nixpkgs=$(pwd)`
- `nix-shell -p python3 python3Packages.pysimplegui`
- `python`
- Then use a code sample on the repl, or call a script
For example:
```python
import PySimpleGUI as sg

layout = [[sg.Button(f'{row}, {col}') for col in range(4)] for row in range(4)]

event, values = sg.Window('List Comprehensions', layout).read(close=True)
```

**Review by thiagokokada — APPROVED** [review]
Looks mostly fine to me, just another question.

There are many variations of `PySimpleGUI`. We eventually may want to support other versions (like the Qt one). So maybe we should call this `PySimpleGUI-tk`? Or since this seems to be the default one, this can be called `PySimpleGUI` and the other ones can be like `PySimpleGUI-qt`.

BTW, how is this package called in other distros (including its variations)?

**Comment by lucasew** [comment]
> Looks mostly fine to me, just another question.
> 
> There are many variations of `PySimpleGUI`. We eventually may want to support other versions (like the Qt one). So maybe we should call this `PySimpleGUI-tk`? Or since this seems to be the default one, this can be called `PySimpleGUI` and the other ones can be like `PySimpleGUI-qt`.
> 
> BTW, how is this package called in other distros (including its variations)?

They doesn't split each variation in it's own package [1]

And for some reason the importing of the other variations doesn't work here, just the main PySimpleGUI

[1] https://repology.org/project/python:pysimplegui/versions

**Comment by thiagokokada** [comment]
> They doesn't split each variation in it's own package [1]

And this: https://pypi.org/project/PySimpleGUIQt/ ?

> And for some reason the importing of the other variations doesn't work here, just the main PySimpleGUI

Isn't it because it is lacking PySide or PyQt (for the Qt side)?

**Comment by lucasew** [comment]
> > They doesn't split each variation in it's own package [1]
> 
> And this: https://pypi.org/project/PySimpleGUIQt/ ?
> 
> > And for some reason the importing of the other variations doesn't work here, just the main PySimpleGUI
> 
> Isn't it because it is lacking PySide or PyQt (for the Qt side)?

It seems he released that separately then decided to join all variations under the same package but was unable to delete the legacy packages.

`import PyAutoGUIQt` is still missing even adding pyqt5 and pyside to the nix-shell

**Comment by thiagokokada** [comment]
> It seems he released that separately then decided to join all variations under the same package but was unable to delete the legacy packages.

I don't think so, at least looking at the [README.md](https://github.com/PySimpleGUI/PySimpleGUI/blob/master/readme.md) they seem to point to this exactly package that I linked above (version 0.35.0 for Qt):
2021-12-30_15-05-39-screenshot

**Comment by lucasew** [comment]
HMMMM

So the canonical version is the Tkinter version, I thought it would ship everything in the same package

So other variations of PySimpleGUI should come on other packages, or PRs

**Comment by thiagokokada** [comment]
> So other variations of PySimpleGUI should come on other packages, or PRs

Yeah, my only question is that if having this as `pysimplegui` because this is the main one and the others as `pysimplegui-qt`, `pysimplegui-wx` is fine, or should we have this called as `pysimplegui-tk` to keep the consistency.

**Comment by lucasew** [comment]
idk for real but upstream names the tk version as just PySimpleGUI

BTW found a thing to test the library

```python
import PySimpleGUI as sg
sg.main()
```

Very cool and useful

**Review by thiagokokada — APPROVED** [review]

File: pkgs/top-level/python-packages.nix [hunk 5]
```diff
@@ -6409,6 +6409,8 @@ in {
 
   pyautogui = callPackage ../development/python-modules/pyautogui { };
 
+  pysimplegui = callPackage ../development/python-modules/pysimplegui { };
```
Comment (thiagokokada) [inline pkgs/top-level/python-packages.nix:6412]: Please put this in alphabetic order.

Comment (lucasew) [reply pkgs/top-level/python-packages.nix]: ok

File: pkgs/development/python-modules/pysimplegui/default.nix [hunk 6]
```diff
@@ -0,0 +1,27 @@
+{ lib
+, buildPythonPackage
+, fetchPypi
+, tkinter
+}:
+buildPythonPackage rec {
+  pname = "PySimpleGUI";
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/pysimplegui/default.nix:7]: pname should match the attribute.

**Pull request merged**